control_toolbox/src/pid.cpp

```diff
@@ -50,46 +50,36 @@ constexpr double UMAX_INFINITY = std::numeric_limits<double>::infinity();
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wdeprecated-declarations"
 Pid::Pid(double p, double i, double d, double i_max, double i_min, bool antiwindup)
-: Pid(
-    p, i, d, i_max, i_min, UMAX_INFINITY, -UMAX_INFINITY, 0.0, antiwindup, AntiwindupStrategy::NONE)
-{
-}
-#pragma GCC diagnostic pop
-
-Pid::Pid(
-  double p, double i, double d, double i_max, double i_min, double u_max, double u_min,
-  double trk_tc, bool antiwindup, AntiwindupStrategy antiwindup_strat)
 : gains_buffer_()
 {
   if (i_min > i_max)
   {
     throw std::invalid_argument("received i_min > i_max");
   }
-  else if (u_min > u_max)
-  {
-    throw std::invalid_argument("received u_min > u_max");
-  }
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
-  set_gains(p, i, d, i_max, i_min, u_max, u_min, trk_tc, antiwindup, antiwindup_strat);
-#pragma GCC diagnostic pop
+  AntiWindupStrategy antiwindup_strat;
```

Comment:
we can remove the i_min check here, as we check it later in 
```
bool Pid::set_gains(
  double p, double i, double d, double u_max, double u_min,
  const AntiWindupStrategy & antiwindup_strat)
```

Reply:
I had to leave that for backward compatibility reasons. Because old method used to throw an exception and now we check but don't throw exceptions. That's why